script/smoke_test.sh

```diff
@@ -3,9 +3,9 @@
 # This is an integration test to generally check the quality of parsing.
 
 SCALA_SCALA_LIBRARY_EXPECTED=95
-SCALA_SCALA_COMPILER_EXPECTED=87
+SCALA_SCALA_COMPILER_EXPECTED=86
```

Comment:
There's a degradation in a single file: 

[StdAttachments.scala#L214](https://github.com/scala/scala/blob/4905801fc161300ba20bae4f7adf045b1698e831/src/compiler/scala/tools/nsc/typechecker/StdAttachments.scala#L214)                 	5 ms	(ERROR [213, 13] - [214, 5]), which is minimized to 
```scala
package x

trait A {
  self: A =>
  import a._
}

trait B {
  self: B =>

  class A() 
  val x = 1
}
```
It seams feasible to me to  investigate the issue individually after this PR gets merged